src/main/scala/com/ubirch/models/Scopes.scala

```diff
@@ -0,0 +1,73 @@
+package com.ubirch.models
+
+import java.net.URL
+
+sealed trait Resource
+object Resource {
+  final val list: List[Resource] = List(UPP, Thing, User)
+  def fromString(value: String): Option[Resource] = {
+    list.find(_.toString.toLowerCase == value)
+  }
+}
+case object UPP extends Resource
+case object Thing extends Resource
+case object User extends Resource
+
+sealed trait Action
+object Action {
+  final val list: List[Action] = List(Anchor, Verify, Create, GetInfo)
+  def fromString(value: String): Option[Action] = {
+    list.find(_.toString.toLowerCase == value)
+  }
+}
+case object Anchor extends Action
+case object Verify extends Action
+case object Create extends Action
+case object GetInfo extends Action
+
+object Scopes {
+
+  type Scope = (Resource, Action)
```

Comment:
why don't use a case class like this?
it's readable for me.
```scala
case class Scope(resource: Resource, action: Action)
```

Reply:
Yes, this is more readable. Added suggestion.